### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by eugenepaniot** [tarantool/tarantool#7582]
**Bug description**

* OS: [e.g. Linux/MacOS/FreeBSD/OpenBSD]
* OS Version: [e.g. Debian 11, FreeBSD 11]
* Architecture: [amd64, arm]

Linux Ubuntu 20.04

```
# ldd /usr/bin/tarantool
	linux-vdso.so.1 (0x00007ffdc3131000)
	libicui18n.so.66 => /lib/x86_64-linux-gnu/libicui18n.so.66 (0x00007f357cae9000)
	libicuuc.so.66 => /lib/x86_64-linux-gnu/libicuuc.so.66 (0x00007f357c903000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f357c8fd000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f357c8e1000)
	libreadline.so.8 => /lib/x86_64-linux-gnu/libreadline.so.8 (0x00007f357c891000)
	libssl.so.1.1 => /lib/x86_64-linux-gnu/libssl.so.1.1 (0x00007f357c7fe000)
	libcrypto.so.1.1 => /lib/x86_64-linux-gnu/libcrypto.so.1.1 (0x00007f357c526000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f357c503000)
	libgomp.so.1 => /lib/x86_64-linux-gnu/libgomp.so.1 (0x00007f357c4c1000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f357c2df000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f357c190000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f357c175000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f357bf81000)
	libicudata.so.66 => /lib/x86_64-linux-gnu/libicudata.so.66 (0x00007f357a4c0000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f357d4fd000)
	libtinfo.so.6 => /lib/x86_64-linux-gnu/libtinfo.so.6 (0x00007f357a490000)
```

Place `tarantool --version` output here.

```
# tarantool --version
Tarantool 2.10.1-0-g482d91c66
Target: Linux-x86_64-RelWithDebInfo
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr -DENABLE_BACKTRACE=ON
Compiler: /usr/bin/cc /usr/lib/ccache/g++
C_FLAGS:-g -O2 -fdebug-prefix-map=/build/tarantool-2.10.1=. -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fexceptions -funwind-tables -fno-common -fopenmp -msse2 -std=c11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-gnu-alignof-expression -fno-gnu89-inline -Wno-cast-function-type
CXX_FLAGS:-g -O2 -fdebug-prefix-map=/build/tarantool-2.10.1=. -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fexceptions -funwind-tables -fno-common -fopenmp -msse2 -std=c++11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-invalid-offsetof -Wno-gnu-alignof-expression -Wno-cast-function-type
```

**Steps to reproduce**

Run two tarantool instances iteractively.

1. Type '...' in console 1.
1. Type '...' in console 2.

or reproducer script:

```
#!/usr/bin/env tarantool

<your code here>
```

How to run:

Just type `tarantool test.lua`.

**Actual behavior**

A clear and concise description of what you see.

```
Segmentation fault
  code: 128
  addr: (nil)
  context: 0x7f99036fed80
  siginfo: 0x7f99036feeb0
  rax      0x558ec113d710     94071613019920
  rbx      0x7f99be11616a     140298295533930
  rcx      0x558ec112d060     94071612952672
  rdx      0x3b               59
  rsi      0x32cd             13005
  rdi      0x7f99be11616b     140298295533931
  rsp      0x7f99036ff340     140295164392256
  rbp      0x7f99990773e1     140297674126305
  r8       0x558ec113d5c0     94071613019584
  r9       0x7f9ae954dd00     140303316344064
  r10      0x7f9ae8b08000     140303305572352
  r11      0x4                4
  r12      0x7f9ae954dd50     140303316344144
  r13      0x3061637469766500 3486176938198197504
  r14      0x2                2
  r15      0x558ec113d638     94071613019704
  rip      0x558ebf623791     94071584602001
  eflags   0x10216            66070
  cs       0x33               51
  gs       0x0                0
  fs       0x0                0
  cr2      0x0                0
  err      0x0                0
  oldmask  0x0                0
  trapno   0xd                13
Current time: 1660747702
Please file a bug at http://github.com/tarantool/tarantool/issues
Attempting backtrace... Note: since the server has already crashed,
this may fail as well
#1  0x558ebf4c02f2 in crash_signal_cb+194
#2  0x7f9b6d0b9420 in __restore_rt+0
#3  0x558ebf623791 in int tuple_compare_slowpath<false, false, false, false>(tuple*, unsigned long, tuple*, unsigned long, key_def*)+193
#4  0x558ebf318cca in NS_USE_HINT::bps_tree_memtx_tree_collect_path(NS_USE_HINT::memtx_tree*, memtx_tree_data<true>, NS_USE_HINT::bpsmemtx_tree_inner_path_elem*, NS_USE_HINT::bpsmemtx_tree_leaf_path_elem*, bool*)+938
#5  0x558ebf32264b in NS_USE_HINT::memtx_tree_insert(NS_USE_HINT::memtx_tree*, memtx_tree_data<true>, memtx_tree_data<true>*, memtx_tree_data<true>*)+91
#6  0x558ebf326228 in int memtx_tree_index_replace<true>(index*, tuple*, tuple*, dup_replace_mode, tuple**, tuple**)+152
#7  0x558ebf344f31 in memtx_space_replace_all_keys+241
#8  0x558ebf3437f3 in memtx_space_execute_upsert+659
#9  0x558ebf3b4f3d in space_execute_dml+909
#10 0x558ebf3eceb7 in box_process_rw+311
#11 0x558ebf3f165d in box_upsert+109
#12 0x558ebf485342 in lbox_upsert+290
#13 0x558ebf4ff4a3 in lj_BC_FUNCC+70
#14 0x558ebAborted (core dumped)
```

**Expected behavior**

**Comment by Totktonada** [comment]
Please, add a script, which will allow us to reproduce the wrong behaviour.

**Comment by kyukhin** [comment]
Could you please give it a try w/ 2.10.2.

**Comment by Totktonada** [comment]
Our current guess that it is another incarnation of #7605 and was fixed in 2.10.2 (only 2.10.1 is affected). Feel free to reopen the issue (with a reproducer script) if the problem will be reproduced on 2.10.2.
